### PR TITLE
[SPARK-47314][DOC] Remove the  wrong comment line of `ExternalSorter#writePartitionedMapOutput` method

### PR DESCRIPTION
**Pull request by zwangsheng**
### What changes were proposed in this pull request?
Correct the comment of `ExternalSorter#writePartitionedMapOutput`.

`ExternalSorter#writePartitionedMapOutput` return nothing, and will update the partition's length when call partition pair writer to close.

### Why are the changes needed?

Correct comment.

### Does this PR introduce _any_ user-facing change?
No, developers will meet this change in source code.

### How was this patch tested?

No need.

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

**Comment by zwangsheng** [comment]
Hi @mridulm @yaooqinn @mccheah, aims to correct the comment, please take a review, thanks.

**Review by yaooqinn — APPROVED** [review]

File: core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala [hunk 1]
```diff
@@ -690,7 +690,7 @@ private[spark] class ExternalSorter[K, V, C](
    * Write all the data added into this ExternalSorter into a map output writer that pushes bytes
    * to some arbitrary backing store. This is called by the SortShuffleWriter.
    *
-   * @return array of lengths, in bytes, of each partition of the file (used by map output tracker)
+   * Update the partition's length, when call partition pair writer to close.
```
Comment (yaooqinn) [inline core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala:693]: we can remove this line?

Comment (LuciferYang) [reply core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala:693]: +1

Comment (zwangsheng) [reply core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala:693]: OK

**Comment by yaooqinn** [comment]
Thanks @zwangsheng， merged to master

**Pull request closed without merging**